tutorials/workflow/csharp/combined-patterns/WorkflowApp/Program.cs

```diff
@@ -0,0 +1,87 @@
+using Dapr.Client;
+using Dapr.Workflow;
+using WorkflowApp;
+using WorkflowApp.Activities;
+
+var builder = WebApplication.CreateBuilder(args);
+builder.Services.AddSingleton<HttpClient>(DaprClient.CreateInvokeHttpClient(appId: "shipping"));
+builder.Services.AddSingleton<InventoryManagement>();
+builder.Services.AddDaprClient();
+builder.Services.AddDaprWorkflow(options => {
+    options.RegisterWorkflow<OrderWorkflow>();
+    options.RegisterActivity<CheckInventory>();
+    options.RegisterActivity<CheckShippingDestination>();
+    options.RegisterActivity<UpdateInventory>();
+    options.RegisterActivity<ProcessPayment>();
+    options.RegisterActivity<ReimburseCustomer>();
+    options.RegisterActivity<RegisterShipment>();
+});
+var app = builder.Build();
+app.UseCloudEvents();
+
+app.MapPost("/start", async (
+    Order order,
+    InventoryManagement inventory) => {
+
+    // This is to ensure to have enough inventory for the order.
+    await inventory.CreateDefaultInventoryAsync();
+
+    await using var scope  = app.Services.CreateAsyncScope();
```

Comment:
Nope, as in a previous message- this isn't necessary. Inject `DaprWorkflowClient` into the async method (can use an optional `[FromServices]` attribute if you want) and access the instance from there. A separate scope is created by invocation of the inbound post request itself and isn't needed here.